frontend/src/ches/MergeForm.jsx

```diff
@@ -628,6 +634,41 @@ class MergeForm extends Component {
       return o;
     };
 
+    // this will suppress a console warning about moment deprecating a defualt fallback on non ISO/RFC2822 date formats
+    // we will just force it to use the new Date constructor.
+    moment.createFromInputFallback = function (config){
+      config._d = new Date(config._i);
+    };
+
+    const parseDelayTs = (value) => {
+      // eslint-disable-next-line no-console
+      console.log(`parseDelayTs value=${value}, isTimestamp? ${moment(value, MOMENT_PARSE_FORMAT, true).format()}`);
+      if (moment(value, MOMENT_PARSE_FORMAT, true).format() !== 'Invalid date') {
+        //return the utc integer value of the timestamp
+        return moment(value).utc().valueOf();
+      }
+      // no good, return undefined so API doesn't process it.
+      return undefined;
+    };
+
+    const handleDefault = (context, data, fieldName, key) =>  {
+      const value = data[key];
+      // use relaxed mode to determine if date (would fail with 0:00 on strict)
+      const isDate = (moment(value, MOMENT_PARSE_FORMAT).format() !== 'Invalid date');
+      // be strict when determining if timestamp - we want valid h:mm
+      const isTimestamp = (moment(value, MOMENT_PARSE_FORMAT, true).format() !== 'Invalid date');
+      const isTimestampField = isTimestamp && fieldName.toLowerCase().endsWith('ts');
+      if (isTimestampField) {
+        const ts = moment(value);
+        context[fieldName] = ts.format(TS_FORMAT);
+      } else if (isDate) {
+        const dt = moment(value);
+        context[fieldName] = dt.format(DT_FORMAT);
+      } else {
+        context[fieldName] = value;
+      }
+    };
```

Comment:
Another thing to refactor to utils potentially, could help with testability

Reply:
👍